src/components/DropZone/components/FileUpload/FileUpload.tsx

```diff
@@ -2,150 +2,98 @@ import * as React from 'react';
 
-import {withAppProvider, WithAppProviderProps} from '../../../AppProvider';
 import Link from '../../../Link';
 import Icon from '../../../Icon';
 import Stack from '../../../Stack';
 import Button from '../../../Button';
 import Caption from '../../../Caption';
 import TextStyle from '../../../TextStyle';
-import withContext from '../../../WithContext';
-import withRef from '../../../WithRef';
+
 import {fileUpload, imageUpload} from '../../images';
-import DropZoneContext, {DropZoneContextType} from '../../context';
 
-import styles from './FileUpload.scss';
+import DropZoneContext from '../../context';
+import {usePolaris} from '../../../../hooks';
 
-export interface State {
-  actionTitle?: string;
-  actionHint?: string;
-}
+import styles from './FileUpload.scss';
 
 export interface Props {
   actionTitle?: string;
   actionHint?: string;
 }
 
-export type CombinedProps = Props &
-  WithAppProviderProps &
-  WithContextTypes<DropZoneContextType>;
-export class FileUpload extends React.Component<CombinedProps, State> {
-  constructor(props: CombinedProps) {
-    super(props);
-
-    const {
-      polaris: {
-        intl: {translate},
-      },
-      context: {type},
-    } = props;
-    const suffix = capitalize(type);
-
-    this.state = {
-      actionTitle: translate(
-        `Polaris.DropZone.FileUpload.actionTitle${suffix}`,
-      ),
-      actionHint: translate(`Polaris.DropZone.FileUpload.actionHint${suffix}`),
-    };
-  }
-
-  updateStateFromProps(props: Props) {
-    const {actionTitle, actionHint} = this.state;
-
-    if (props.actionTitle && props.actionTitle !== actionTitle) {
-      this.setState({actionTitle: props.actionTitle});
-    }
-
-    if (props.actionHint && props.actionHint !== actionHint) {
-      this.setState({actionHint: props.actionHint});
-    }
-  }
-
-  componentWillReceiveProps(props: Props) {
-    this.updateStateFromProps(props);
-  }
-
-  componentWillMount() {
-    this.updateStateFromProps(this.props);
-  }
-
-  render() {
-    const {
-      context: {size, type},
-    } = this.props;
-    const {actionTitle, actionHint} = this.state;
-    const imageClasses = classNames(
-      styles.Image,
-      size && size === 'extraLarge' && styles.sizeExtraLarge,
-      size && size === 'large' && styles.sizeLarge,
-    );
-
-    const extraLargeView =
-      size === 'extraLarge' ? (
-        <Stack vertical>
-          {type === 'file' && (
-            <img className={imageClasses} src={fileUpload} alt="" />
-          )}
-          {type === 'image' && (
-            <img className={imageClasses} src={imageUpload} alt="" />
-          )}
-          <Button>{actionTitle}</Button>
+function FileUpload(props: Props) {
+  const {
+    intl: {translate},
+  } = usePolaris();
+  const {size, type} = React.useContext(DropZoneContext);
+  const suffix = capitalize(type);
+  const {
+    actionTitle = translate(`Polaris.DropZone.FileUpload.actionTitle${suffix}`),
```

Comment:
We use the provided actionTitle/Hint and default it to what was previously the default on state

Reply:
I wonder why we had done that?

Reply:
Seems like it was to avoid string translation however object lookup is inexpensive so I'm not sure why, or this approach didn't occur when developing the component? 🤷‍♂